algorithms/dfs/maze_search.py

```diff
@@ -0,0 +1,62 @@
+'''
+Find shortest path from top left column to the right lowest column using DFS.
+only step on the columns whose value is 1
+if there is no path, it returns -1
+
+Ex 1)
+If maze is
+[[1,0,1,1,1,1],
+ [1,0,1,0,1,0],
+ [1,0,1,0,1,1],
+ [1,1,1,0,1,1]],
+the answer is: 14
+
+Ex 2)
+If maze is
+[[1,0,0],
+ [0,1,1],
+ [0,1,1]],
+the answer is: -1
+'''
+
+global cnt
+global directions
```

Comment:
please don't use global variables
(including directions below)

also this code requires unittests